api/api.proto

```diff
@@ -2684,3 +2698,53 @@ message LocateResponse {
     // <containerid>: /var/www
     map<string, string> dockerids = 2;
 }
+
+message VolumePlacementSpec {
+  // Defines a list of rules as part of the placement spec
+  repeated VolumePlacementRule rules = 1;
+}
+
+message VolumePlacementRule {
```

Comment:
The messages themselves need at least a small brief comment. This one may want to explain what happens if you combine different optional values. If you cannot combine them, then they should go into a `oneof`.

Reply:
Added comment above here

```
message VolumePlacementStrategy {
  // Defines a list of rules as part of the placement spec. All the rules specified will
  // be applied for volume placement.
  // Rules that have enforcement as "required" are strictly enforced while "preferred" are best effort.
  // In situations, where 2 or more rules conflict, the weight of the rules will dictate which wins.
  repeated VolumePlacementRule rules = 1;
}
```

Reply:
It's not a oneof since we have use cases to have both. Also I want the spec to allow multiple rules. An implementation can fail the request when parsing the strategy if it doesn't support it.